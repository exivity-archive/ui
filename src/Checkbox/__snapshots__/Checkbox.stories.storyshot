// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots interact|Checkbox default 1`] = `
.c0 {
  padding: 12px;
}

.c4 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c4 > .c3 {
  display: block;
  font-size: 0.875em;
  color: #999;
}

.c2 {
  position: absolute;
  opacity: 0;
  z-index: -1;
}

.c2 + label {
  position: relative;
  display: inline-block;
  padding: 0 0 0 2em;
  cursor: pointer;
  line-height: 1em;
  height: 1em;
  color: #999;
  will-change: color;
  -webkit-transition: 0.25s color ease;
  transition: 0.25s color ease;
}

.c2 + label:focus {
  outline: 0;
}

.c2 + label:empty {
  padding: 0 0 0 1em;
}

.c2 + label::before,
.c2 + label::after {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  width: 1em;
  height: 1em;
  box-sizing: border-box;
  will-change: transform,background-color,color;
  -webkit-transition: 0.25s all ease;
  transition: 0.25s all ease;
}

.c2 + label::before {
  content: " ";
  border-radius: 4px;
  background-color: #eee;
}

.c2 + label::after {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c2 + label::after {
  content: "\\2714";
  color: #fff;
  line-height: 1.1em;
  text-align: center;
}

.c2:focus + label::before,
.c2 + label:hover::before {
  background-color: #999;
}

.c2:checked + label {
  color: #222;
}

.c2:checked + label::before {
  background-color: #222;
}

.c2:checked + label::after {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c1 {
  margin: 1.25rem 0;
}

.c1:first-child {
  margin-top: 0;
}

.c1:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
>
  <section
    className="c1"
  >
    <span>
      <input
        checked={true}
        className="c2"
        id="0"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c3 c4"
        htmlFor="0"
      >
        
      </label>
    </span>
  </section>
  <section
    className="c1"
  >
    <span>
      <input
        checked={false}
        className="c2"
        id="1"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c3 c4"
        htmlFor="1"
      >
        
      </label>
    </span>
  </section>
</div>
`;

exports[`Storyshots interact|Checkbox with custom label 1`] = `
.c0 {
  padding: 12px;
}

.c1 {
  position: absolute;
  opacity: 0;
  z-index: -1;
}

.c1 + label {
  position: relative;
  display: inline-block;
  padding: 0 0 0 2em;
  cursor: pointer;
  line-height: 1em;
  height: 1em;
  color: #999;
  will-change: color;
  -webkit-transition: 0.25s color ease;
  transition: 0.25s color ease;
}

.c1 + label:focus {
  outline: 0;
}

.c1 + label:empty {
  padding: 0 0 0 1em;
}

.c1 + label::before,
.c1 + label::after {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  width: 1em;
  height: 1em;
  box-sizing: border-box;
  will-change: transform,background-color,color;
  -webkit-transition: 0.25s all ease;
  transition: 0.25s all ease;
}

.c1 + label::before {
  content: " ";
  border-radius: 4px;
  background-color: #eee;
}

.c1 + label::after {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c1 + label::after {
  content: "\\2714";
  color: #fff;
  line-height: 1.1em;
  text-align: center;
}

.c1:focus + label::before,
.c1 + label:hover::before {
  background-color: #999;
}

.c1:checked + label {
  color: #222;
}

.c1:checked + label::before {
  background-color: #222;
}

.c1:checked + label::after {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  margin: 1.25rem 0;
}

.c2:first-child {
  margin-top: 0;
}

.c2:last-child {
  margin-bottom: 0;
}

.c2,
input.c2 {
  font-size: 1.5em;
}

.c2 .c3 {
  margin-right: 1.25em;
  display: inline-block;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
  -webkit-transform-origin: left 25%;
  -ms-transform-origin: left 25%;
  transform-origin: left 25%;
}

<div
  className="c0"
>
  <span>
    <input
      checked={true}
      className="c1"
      id="3"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="c2"
      htmlFor="3"
      style={
        Object {
          "marginTop": 0,
        }
      }
      type="header"
    >
      Check me out
    </label>
  </span>
</div>
`;

exports[`Storyshots interact|Checkbox with label 1`] = `
.c0 {
  padding: 12px;
}

.c3 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c3 > .c2 {
  display: block;
  font-size: 0.875em;
  color: #999;
}

.c1 {
  position: absolute;
  opacity: 0;
  z-index: -1;
}

.c1 + label {
  position: relative;
  display: inline-block;
  padding: 0 0 0 2em;
  cursor: pointer;
  line-height: 1em;
  height: 1em;
  color: #999;
  will-change: color;
  -webkit-transition: 0.25s color ease;
  transition: 0.25s color ease;
}

.c1 + label:focus {
  outline: 0;
}

.c1 + label:empty {
  padding: 0 0 0 1em;
}

.c1 + label::before,
.c1 + label::after {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  width: 1em;
  height: 1em;
  box-sizing: border-box;
  will-change: transform,background-color,color;
  -webkit-transition: 0.25s all ease;
  transition: 0.25s all ease;
}

.c1 + label::before {
  content: " ";
  border-radius: 4px;
  background-color: #eee;
}

.c1 + label::after {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c1 + label::after {
  content: "\\2714";
  color: #fff;
  line-height: 1.1em;
  text-align: center;
}

.c1:focus + label::before,
.c1 + label:hover::before {
  background-color: #999;
}

.c1:checked + label {
  color: #222;
}

.c1:checked + label::before {
  background-color: #222;
}

.c1:checked + label::after {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

<div
  className="c0"
>
  <span>
    <input
      checked={true}
      className="c1"
      id="2"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="c2 c3"
      htmlFor="2"
    >
      Enable this setting
    </label>
  </span>
</div>
`;
