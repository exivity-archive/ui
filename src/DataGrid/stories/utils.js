/**
 * Generate a unique random 8 character id (not cryptographically secure).
 *
 * @returns {string}
 */
const randomIdMemory = []
export const randomId = () => {
  let id

  if (process.env.NODE_ENV !== 'test') {
    do {
      id = Math.random().toString(36).substring(2, 10)
    } while (randomIdMemory.indexOf(id) > -1)
  } else {
    id = randomIdMemory.length
  }

  randomIdMemory.push(id)

  return id
}

/* eslint-disable-next-line max-len */
const RANDOM_WORDS = ['abstrusity', 'advertisable', 'bellwood', 'benzole', 'boreum', 'brenda', 'cassiopeian', 'chansonnier', 'cleric', 'conclusional', 'conventicle', 'copalm', 'cornopion', 'crossbar', 'disputative', 'djilas', 'ebracteate', 'ephemerally', 'epidemical', 'evasive', 'eyeglasses', 'farragut', 'fenny', 'ferryman', 'fluently', 'foreigner', 'genseng', 'glaiket', 'haunch', 'histogeny', 'illocution', 'imprescriptible', 'inapproachable', 'incisory', 'intrusiveness', 'isoceraunic', 'japygid', 'juiciest', 'jump', 'kananga', 'leavening', 'legerdemain', 'licence', 'licia', 'luanda', 'malaga', 'mathewson', 'nonhumus', 'nonsailor', 'nummary', 'nyregyhza', 'onanist', 'opis', 'orphrey', 'paganising', 'pebbling', 'penchi', 'photopia', 'pinocle', 'principally', 'prosector.', 'radiosensitive', 'redbrick', 'reexposure', 'revived', 'subexternal', 'sukarnapura', 'supersphenoid', 'tabularizing', 'territorialism', 'tester', 'thalassography', 'tuberculise', 'uncranked', 'undersawyer', 'unimpartible', 'unsubdivided', 'untwining', 'unwaived', 'webfoot', 'wedeling', 'wellingborough', 'whiffet', 'whipstall', 'wot', 'yonkersite', 'zonary']

function createRandomizedItems (depth, flat = true, parent) {
  let items = []
  let randomName
  let randomNumber
  let numChildren
  let randomValue

  if (process.env.NODE_ENV !== 'test') {
    randomName = RANDOM_WORDS[Math.floor(Math.random() * RANDOM_WORDS.length)]
    randomNumber = Math.floor(Math.random() * 100)
    numChildren = (depth < 3 && flat === false) ? Math.floor(Math.random() * 5) : 0
    randomValue = parseInt(Math.random() * 10, 10)
  } else {
    randomName = 'Test name'
    randomNumber = 50
    numChildren = (depth < 3 && flat === false) ? 3 : 0
    randomValue = 5
  }

  const random = randomId()

  const item = {
    id: random,
    parent: parent || null,
    name: randomName,
    number: randomNumber,
    group: 'Group ' + randomName.substr(0, 1),
    value: randomValue
  }

  items.push(item)

  for (let i = 0; i < numChildren; i++) {
    items = items.concat(createRandomizedItems(depth + 1, flat, item.key))
  }

  return items
}

export function createRandomizedData (items = 100, flat = true) {
  let data = []

  for (let i = 0; i < items; i++) {
    data = data.concat(createRandomizedItems(0, flat))
  }

  return data
}
