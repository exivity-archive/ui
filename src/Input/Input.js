import PropTypes from 'prop-types'
import { Input as BaseInput, styled } from 'reakit'
import { theme } from 'styled-tools'
import withEnumProps from '../withEnumProps'

const Input = styled(BaseInput)`
  border-radius: ${theme('base.borderRadius')};
`

Input.propTypes = {
  ...Input.propTypes,

  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  warning: PropTypes.bool,
  danger: PropTypes.bool,

  scale: PropTypes.number,
  xsmall: PropTypes.bool,
  small: PropTypes.bool,
  large: PropTypes.bool,
  xlarge: PropTypes.bool
}

Input.defaultProps = {
  palette: 'grayscale',
  opaque: false,
  scale: 1
}

export default withEnumProps(Input, {
  palette: 'key',
  scale: 'value'
})
