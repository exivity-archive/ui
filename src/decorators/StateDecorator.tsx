import React from 'react'
import StateContainer from './StateContainer'

export const withState = (initialState = {}) => (storyFunc: any, context: any) => (
  <StateContainer initialState={initialState}>
    {(state: any, storeState: any) => {
      context.state = state
      context.storeState = storeState
      return storyFunc()
    }}
  </StateContainer>
)
