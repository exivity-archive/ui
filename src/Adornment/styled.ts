import styled, { css } from 'styled-components'

import { SizesProps, matchThemeProp } from '../utils/styled'
import { Position } from './Adornment'
import { Block } from '../Block'

export interface StyledAdornmentProps {
  position: Position
  inset: number | string
}

export const StyledAdornment = styled.span <StyledAdornmentProps>`
  display: flex;
  align-items: center;
  position: absolute;
  height: 100%;
  top: 0;
  font-size: ${
  matchThemeProp(
    theme => theme.global.sizes,
    { modifier: (em: number) => em * 20 }
  )}px;

  ${props => props.position === Position.LEFT
    ? css`left: ${props.inset}px;`
    : css`right: ${props.inset}px;`
  }
`

export const AdornmentWrapper = styled(Block)`
  position: relative;
  display: inline-block;
`
