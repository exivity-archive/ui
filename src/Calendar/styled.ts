import styled, { css } from 'styled-components'
import { fromTheme, globalFont } from '../utils/styled'

export const StyledWeekDays = styled.ul`
  ${globalFont}
  opacity: 0.6;
  margin: 0;
  padding: 10px 0;

  li {
    display: inline-block;
    width: 14.28%;
    text-align: center;
  }
`

export const StyledDays = styled.ul`
  ${globalFont}

  padding: 10px 0;
  margin: 0;

  li {
    list-style-type: none;
    display: inline-block;
    width: calc(14.28% - 5px);
    text-align: center;
    margin-right: 5px;
    margin-bottom: 5px;
  }
`

export const StyledDay = styled.div<{ active: boolean }>`
  height: 100%;
  border-radius: 180px;
  padding: 5px;
  cursor: pointer;
  user-select: none;

  &:hover {
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  }

  ${({ active }) => active && css`
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  `}
`
