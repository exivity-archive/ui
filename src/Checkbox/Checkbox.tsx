import React, { ReactElement, ReactNode, useMemo } from 'react'

import styled, { css } from 'styled-components'
import { OmitOnChangeHTMLInputAttributes, OnChange } from '../Input/Input'
import { Label } from '../Label'
import { randomId } from '../utils/randomId'
import { fromTheme, StyledProps } from '../utils/styled'

export interface CheckboxProps extends StyledProps, OmitOnChangeHTMLInputAttributes {
  radio?: boolean
  checked?: boolean
  onChange?: OnChange<boolean>
  label?: string | ReactNode
}

const animationSpeed = '0.25s'

export const StyledCheckbox = styled.input`
  // Take it out of document flow and hide it
  position: absolute;
  opacity: 0;
  z-index: -1;

  & + label {
    position: relative;
    display: inline-block;
    padding: 0 0 0 2em;
    cursor: pointer;
    line-height: 1em;
    height: 1em;

    color: ${fromTheme(theme => theme.global.textColorMuted)};

    will-change: color;
    transition: ${animationSpeed} color ease;

    &:focus {
      outline: 0;
    }

    &:empty {
      padding: 0 0 0 1em;
    }

    &::before,
    &::after {
      position: absolute;
      top: 0;
      left: 0;
      display: block;
      width: 1em;
      height: 1em;
      box-sizing: border-box;
      will-change: transform, background-color, color;
      transition: ${animationSpeed} all ease;
    }

    &::before {
      content: " ";
      border-radius: ${props => props.type === 'checkbox'
        ? css`${fromTheme(theme => theme.global.borderRadius)}px`
        : '50%'};
      background-color: ${fromTheme(theme => theme.colors.lightGray)};
    }

    &::after {
      transform: scale(0);
    }

    ${props => props.type === 'checkbox' && css`
      &::after {
        content: "\\2714";
        color: ${fromTheme(theme => theme.colors.white)};
        line-height: 1.1em;
        text-align: center;
      }
    `}

    ${props => props.type === 'radio' && css`
      &::after {
        content: " ";
        background: ${fromTheme(theme => theme.colors.dark)};
        border: .2em solid ${fromTheme(theme => theme.colors.lightGray)};
        border-radius: 50%;
      }
    `}
  }

  &:focus + label::before,
  & + label:hover::before {
    background-color: ${fromTheme(theme => theme.colors.gray)};
  }

  &:checked + label {
    color: ${fromTheme(theme => theme.global.textColor)};

    &::before {
      background-color: ${fromTheme(theme => theme.colors.dark)};
    }

    &::after {
      transform: scale(0.9);
    }
  }
`

export const Checkbox = ({ radio, checked, onChange, label = '', id, ...props }: CheckboxProps) => {
  const memoizedId = useMemo(() => id || randomId(), [])

  return <span>
    <StyledCheckbox
      type={radio ? 'radio' : 'checkbox'}
      onChange={event => {
        onChange && onChange(event.target.checked, event)
      }}
      checked={checked}
      id={memoizedId}
      {...props}
    />
    {typeof label === 'string'
      ? <Label htmlFor={memoizedId}>{label}</Label>
      : React.cloneElement(label as ReactElement, {
        as: 'label',
        htmlFor: memoizedId
      })}
  </span>
}
