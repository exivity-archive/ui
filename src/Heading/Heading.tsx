import React, { ReactNode } from 'react'
import styled, { css } from 'styled-components'
import { Icon } from '../Icon'
import { SectionProps, fromTheme, globalSectionSpacing, globalFont, StyledProps } from '../utils/styled'
import { preciseEm } from '../utils/styled/isolated'

enum Levels {
  'header' = 1,
  'section' = 2,
  'sub' = 3
}

export type HeadingType = 'header' | 'section' | 'sub'

export interface HeadingProps extends StyledProps, SectionProps {
  type?: HeadingType
  children?: ReactNode
}

export const StyledHeading = styled.div<HeadingProps>`
  ${globalFont};
  ${globalSectionSpacing};

  // Add extra specificity when rendered as input
  &, input& {
    ${(props: HeadingProps) => props.type === 'header' && css`
      font-size: ${fromTheme(theme => theme.global.sizes.huge)}em;
    `}

    ${(props: HeadingProps) => props.type === 'section' && css`
      font-size: ${fromTheme(theme => theme.global.sizes.huge)}em;
      color: ${fromTheme(theme => theme.global.purposes.primary)};
    `}

    ${(props: HeadingProps) => props.type === 'sub' && css`
      font-size: ${fromTheme(theme => theme.global.sizes.small)}em;
      text-transform: uppercase;
      color: ${fromTheme(theme => theme.colors.gray)};
      display: table;
      overflow: hidden;
      white-space: nowrap;

      &:after {
        border-top: 1px solid ${fromTheme(theme => theme.colors.lightGray)};
        content: '';
        display: table-cell;
        position: relative;
        top: ${preciseEm(0.6)}em;
        width: 100%;
      }

      &:after { left: 1.5%; }
    `}

  }

  ${Icon} {
    margin-right: ${fromTheme(theme => theme.global.baseSpacing)}em;
    display: inline-block;
    transform: scale(1.5);
    transform-origin: left 25%; // 25% from trial and error
  }
`
export const Heading = ({ type = 'header', ...rest }: HeadingProps) => (
  <StyledHeading
    as={`h${Levels[type]}` as 'h1' | 'h2' | 'h3'}
    type={type}
    {...rest}
  />
)

Heading.displayName = 'Heading'
