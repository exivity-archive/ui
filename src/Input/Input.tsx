import React, { ChangeEvent, InputHTMLAttributes, useState, forwardRef, Ref, useRef, Dispatch } from 'react'

import { Omit } from '../utils/types'
import { StyledInputProps, StyledContainer, AnimatedStyledInput } from './styled'

const makeHandleChange = (onChange: OnChange | undefined, setValid: Dispatch<boolean>) => {
  return (event: ChangeEvent<HTMLInputElement>) => {
    if (onChange) {
      onChange(event.target.value, event)
      event.target.checkValidity && setValid(event.target.checkValidity())
    }
  }
}

export type OmitOnChangeHTMLInputAttributes = Omit<InputHTMLAttributes<HTMLInputElement>, 'onChange'>

export type OnChange<T = string, E = HTMLInputElement> = (value: T, event: ChangeEvent<E>) => void

export interface InputProps extends StyledInputProps, OmitOnChangeHTMLInputAttributes {
  value?: string | number
  onChange?: OnChange
  required?: boolean
  ref?: React.RefObject<HTMLInputElement> | null
  disabled?: boolean
}

interface Props extends InputProps {
  step?: number | string
  type?: string
}

export const Input =
  forwardRef(({
    type = 'text',
    onChange,

    // Container props
    onClick,
    width,

    // Container and Input props
    inline,

    // Rest goes to Input
    ...rest
  }: Props, ref: Ref<HTMLInputElement>) => {
    const [valid, setValid] = useState(true)

    const containerRef = useRef<HTMLDivElement>(null)

    const container = { onClick, width, inline }

    return (
      <StyledContainer ref={containerRef} {...container}>
        <AnimatedStyledInput
          inline={inline}
          ref={ref}
          type={type}
          danger={!valid}
          onChange={makeHandleChange(onChange, setValid)}
          {...rest} />
      </StyledContainer>
    )
  })
