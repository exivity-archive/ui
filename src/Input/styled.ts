import styled, { css } from 'styled-components'
import { textAlign, TextAlignProps } from 'styled-system'

import { globalFont, matchThemeProp, fromTheme, hexToString, PurposesProps, SizesProps, StyledProps } from '../utils/styled'
import { Icon } from '../Icon'
import { Block } from '../Block'
import { animated } from 'react-spring'

interface ContainerProps {
  inline?: boolean
}

export const StyledContainer = styled.div<ContainerProps>`
  position: relative;

  ${props => (props.inline) && css`
    display: inline;
  `}

  ${Icon} {
    pointer-events: none;
  }
`

export interface StyledInputProps extends PurposesProps, SizesProps, StyledProps, TextAlignProps {
  // Variants
  outlined?: boolean
  flat?: boolean

  // Layout
  inline?: boolean
}

export const inputStyles = css<StyledInputProps>`
  ${globalFont};
  ${textAlign};

  font-size: ${matchThemeProp(theme => theme.global.sizes)}rem;

  box-sizing: border-box;
  padding: calc(0.5em - ${fromTheme(theme => theme.global.borderWidth)}px) 0.5em; // subtract border to get a height of exactly 2.5em for single line items

  border-radius: ${fromTheme(theme => theme.global.borderRadius)}px;
  outline: 0;
  border: 0;

  --focus-color: ${matchThemeProp(theme => theme.global.purposes, { modifier: hexToString })};

  ${props => (!props.inline) && css`
    display: block;
    width: 100%;
  `}

  ${props => (!props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.lightGray)};
    background-color: ${fromTheme(theme => theme.colors.lightGray)};

    &:hover {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
    }

    &:focus {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
    }
  `}

  ${props => (props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${matchThemeProp(theme => theme.global.purposes)};
    background-color: unset;

    &:hover {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.gray)};
    }

    &:focus {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.dark)};
    }
  `}

  ${props => props.flat && css<StyledInputProps>`
    padding: 0;

    ${props => props.outlined && css`
      outline-offset: 9px;

      &:hover {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
      }

      &:focus {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
      }
    `}
  `}

  &::placeholder {
    color: currentcolor;
    opacity: 0.5;
  }

  &[disabled] {
    cursor: not-allowed;
    box-shadow: inset 0 0 999em rgba(128, 128, 128, 0.2);
  }
`

export const StyledInput = styled.input<StyledInputProps>`
  ${inputStyles}
`

export const AnimatedStyledInput = animated(StyledInput)
