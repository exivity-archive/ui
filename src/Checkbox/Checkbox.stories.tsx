import React from 'react'
import { storiesOf } from '@storybook/react'
import { Heading } from '../Heading'
import { TextInput } from '../TextInput'
import { withState } from '../utils/tests/decorators/StateDecorator'
import { storeAndAction } from '../utils/tests/storeAndAction'
import { Checkbox } from '.'

storiesOf('interact|Checkbox', module)
  .addDecorator(withState(true))
  .add('default', ({ state, storeState }: any) => (
    <Checkbox checked={state} onChange={storeAndAction(storeState, 'onChange')} />
  ))
  .add('with label', ({ state, storeState }: any) => (
    <Checkbox checked={state} onChange={storeAndAction(storeState, 'onChange')} label='Enable this setting' />
  ))
  .add('with custom label', ({ state, storeState }: any) => (
    <Checkbox checked={state} onChange={storeAndAction(storeState, 'onChange')} label={
      <Heading style={{ marginTop: 0 }}>Enable this setting</Heading>
    } />
  ))
