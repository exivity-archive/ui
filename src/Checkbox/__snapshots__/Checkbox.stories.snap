// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots interact|Checkbox default 1`] = `
.c0 {
  padding: 12px;
}

.c1 {
  margin: 20px;
  width: 100px;
  height: 100px;
  overflow: hidden;
}

.c1:after {
  position: relative;
  left: 5px;
  top: -30px;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  border: solid #fff;
  border-width: 0 3px 3px 0;
  content: ' ';
  display: block;
  width: 6px;
  height: 12px;
  cursor: pointer;
  pointer-events: none;
}

.c2 {
  width: 20px;
  height: 20px;
  cursor: pointer;
  background: rgba(40,40,40,0.7);
  color: black;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  position: relative;
  left: -5px;
  top: -5px;
}

.c2:focus {
  outline: 0;
}

<div
  className="c0"
>
  <div
    checked={true}
    className="c1"
  >
    <input
      checked={true}
      className="c2"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
